src/parameterizations/lateral/MOM_mixed_layer_restrat.F90

```diff
@@ -968,6 +975,23 @@ subroutine mixedlayer_restrat_Bodner(CS, G, GV, US, h, uhtr, vhtr, tv, forces, d
     wpup(i,j) = rmean2ts(wpup(i,j), CS%wpup_filtered(i,j), &
                          CS%BLD_growing_Tfilt, CS%BLD_decaying_Tfilt, dt)
     CS%wpup_filtered(i,j) = wpup(i,j)
+
+    if (CS%id_lfbod > 0) then
+      ! Calculate front length used in B22 formula (eq 24).
+
+      w_star3 = max(0., -bflux(i,j)) * BLD(i,j)
+      u_star3 = U_star_2d(i,j)**3
+
+      ! Include an absurdly_small_freq2 to prevent division by zero.
+      f2_h = max( &
+          (0.25*((G%CoriolisBu(I,J)  + G%CoriolisBu(I-1,J-1)) &
+            + (G%CoriolisBu(I-1,J) + G%CoriolisBu(I,J-1))))**2, &
+          absurdly_small_freq2)
+
+      lf_bodner_diag(i,j) = &
+          0.25 * cuberoot(CS%mstar * u_star3 + CS%nstar * w_star3)**2 &
+            / (f2_h * max(little_h(i,j), GV%Angstrom_H))
```

Comment:
In order to convert this to units of `[L ~> m]`, the denominator should be multiplied by `(US%L_to_Z * GV%H_to_RZ* tv%SpV_avg(i,j,1))` when in fully non-Boussinesq mode or in Boussinesq mode the denominator should be multiplied by `(GV%H_to_Z*US%L_to_Z)`.  (Compare this with the expressions that set wpup at lines 943 and 968.)  Were we to make this change here, the conversion factor at line 1814 would need to be changed to `US%L_to_m`, while the units of `lf_bodner_diag` on line 811 would stay as they are.

Reply:
I have implemented this suggestion.